x-pack/plugins/maps/public/licensed_features.ts

```diff
@@ -36,13 +36,20 @@ export const LICENCED_FEATURES_DETAILS: Record<LICENSED_FEATURES, LicensedFeatur
 
 let licenseId: string | undefined;
 let isGoldPlus: boolean = false;
-
 let isEnterprisePlus: boolean = false;
 
 export const getLicenseId = () => licenseId;
 export const getIsGoldPlus = () => isGoldPlus;
 
+let initializeLicense: (value: unknown) => void;
```

Comment:
nit: how about moving initializeLicense, licenseInitialized, and whenLicenseInitialized next to setLicensingPluginStart so all of the logic sits in a single place in the file.